tools/testing/gatekeeper/jobs.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2024 Red Hat Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Keeps checking the current PR until all required jobs pass
+Env variables:
+* REQUIRED_JOBS: comma separated list of required jobs (in form of
+                 "$workflow / $job")
+* REQUIRED_REGEXPS: comma separated list of regexps for required jobs
+* COMMIT_HASH: Full commit hash we want to be watching
+* GITHUB_REPOSITORY: Github repository (user/repo)
+Sample execution (GH token can be excluded):
+GITHUB_TOKEN="..." REQUIRED_JOBS="skipper / skipper"
+REQUIRED_REGEXPS=".*" REQUIRED_LABELS="ok-to-test;bar"
+COMMIT_HASH=b8382cea886ad9a8f77d237bcfc0eba0c98775dd
+GITHUB_REPOSITORY=kata-containers/kata-containers
+PR_NUMBER=123 python3 jobs.py
+"""
+
+import os
+import re
+import sys
+import time
+import requests
+
+
+PASS = 0
+FAIL = 1
+RUNNING = 127
+
+
+_GH_HEADERS = {"Accept": "application/vnd.github.v3+json"}
+if os.environ.get("GITHUB_TOKEN"):
+    _GH_HEADERS["Authorization"] = f"token {os.environ['GITHUB_TOKEN']}"
+_GH_API_URL = f"https://api.github.com/repos/{os.environ['GITHUB_REPOSITORY']}"
+_GH_RUNS_URL = f"{_GH_API_URL}/actions/runs"
+
+
+class Checker:
+    """Object to keep watching required GH action workflows"""
+    def __init__(self):
+        self.latest_commit_sha = os.getenv("COMMIT_HASH")
+        self.pr_number = os.getenv("PR_NUMBER")
+        required_labels = os.getenv("REQUIRED_LABELS")
+        if required_labels:
+            self.required_labels = set(required_labels.split(";"))
+        else:
+            self.required_labels = []
+        required_jobs = os.getenv("REQUIRED_JOBS")
+        if required_jobs:
+            required_jobs = required_jobs.split(";")
+        else:
+            required_jobs = []
+        required_regexps = os.getenv("REQUIRED_REGEXPS")
+        self.required_regexps = []
+        # TODO: Add way to specify minimum amount of tests
+        # (eg. via \d+: prefix) and check it in status
+        if required_regexps:
+            for regexp in required_regexps.split(";"):
+                self.required_regexps.append(re.compile(regexp))
+        if not required_jobs and not self.required_regexps:
+            raise RuntimeError("No REQUIRED_JOBS or REQUIRED_REGEXPS defined")
+        # Set all required jobs as RUNNING to enforce waiting for them
```

Comment:
I'm seeing the status `EXPECTED` below. The comment here is a bit confusing. Am I missing something? Thanks!

Reply:
sure, let me change the comment...